### ISSUE TITLE
Docker Desktop 4.13.0 (89412): Unable to pull `postgres:11.2`

### ISSUE THREAD

**Issue by joschi** [docker/for-mac#6540]
- [x] I have tried with the latest version of Docker Desktop
- [x] I have tried disabling enabled experimental features
- [ ] I have uploaded Diagnostics
- Diagnostics ID:

### Expected behavior

Docker Desktop 4.12.0 (85629)

<details>
<summary>docker version</summary>

```
Client:
 Cloud integration: v1.0.29
 Version:           20.10.17
 API version:       1.41
 Go version:        go1.17.11
 Git commit:        100c701
 Built:             Mon Jun  6 23:04:45 2022
 OS/Arch:           darwin/arm64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.12.0 (85629)
 Engine:
  Version:          20.10.17
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.11
  Git commit:       a89b842
  Built:            Mon Jun  6 23:01:01 2022
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.8
  GitCommit:        9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```
</details>

<details>
<summary>docker info</summary>

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.9.1)
  compose: Docker Compose (Docker Inc., v2.10.2)
  extension: Manages Docker extensions (Docker Inc., v0.2.9)
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
  scan: Docker Scan (Docker Inc., v0.19.0)

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 6
 Server Version: 20.10.17
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc version: v1.1.4-0-g5fd4c4d
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.10.124-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 4
 Total Memory: 7.653GiB
 Name: docker-desktop
 ID: VPV5:HJU7:TH4D:7EQ4:Z5Z6:S5HS:3IVP:4ALF:DLXC:GQ2D:I3OK:VR7I
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5000
  127.0.0.0/8
 Live Restore Enabled: false
```
</details>

```
# echo -e "POST /images/create?fromImage=postgres%3A11.2-alpine HTTP/1.0\r\n" | nc -U /var/run/docker.sock
HTTP/1.0 200 OK
Api-Version: 1.41
Content-Type: application/json
Date: Thu, 27 Oct 2022 19:51:52 GMT
Docker-Experimental: false
Ostype: linux
Server: Docker/20.10.17 (linux)

{"status":"Pulling from library/postgres","id":"11.2-alpine"}
[...]
{"status":"Pull complete","progressDetail":{},"id":"dd936443c289"}
{"status":"Digest: sha256:ac1494b4cd5c0c6a85209e645fbb84ea186b55fa8c3678ef66dd9e4edf2b0195"}
{"status":"Status: Downloaded newer image for postgres:11.2-alpine"}
```

### Actual behavior

- Docker Desktop 4.13.0 (89412)
- Docker Desktop 4.13.1 (90346) (tested on 2022-11-02)
- Docker Desktop 4.14.0 (91374) (tested on 2022-11-11)

<details>
<summary>docker version</summary>

```
Client:
 Cloud integration: v1.0.29
 Version:           20.10.20
 API version:       1.41
 Go version:        go1.18.7
 Git commit:        9fdeb9c
 Built:             Tue Oct 18 18:20:35 2022
 OS/Arch:           darwin/arm64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.13.0 (89412)
 Engine:
  Version:          20.10.20
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.7
  Git commit:       03df974
  Built:            Tue Oct 18 18:18:16 2022
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.8
  GitCommit:        9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```
</details>

<details>
<summary>docker info</summary>

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.9.1)
  compose: Docker Compose (Docker Inc., v2.12.0)
  dev: Docker Dev Environments (Docker Inc., v0.0.3)
  extension: Manages Docker extensions (Docker Inc., v0.2.13)
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
  scan: Docker Scan (Docker Inc., v0.21.0)

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 6
 Server Version: 20.10.20
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc version: v1.1.4-0-g5fd4c4d
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.15.49-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 4
 Total Memory: 7.668GiB
 Name: docker-desktop
 ID: PJAL:MTDP:TPRG:RO37:4FSR:REXP:ANEJ:RWJ5:MR2U:3KK5:WJW4:OLJT
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5000
  127.0.0.0/8
 Live Restore Enabled: false
```
</details>

```
# echo -e "POST /images/create?fromImage=postgres%3A11.2-alpine HTTP/1.0\r\n" | nc -U /var/run/docker.sock
HTTP/1.0 500 Internal Server Error
Content-Type: text/plain; charset=utf-8
X-Content-Type-Options: nosniff
Date: Thu, 27 Oct 2022 20:02:46 GMT
Content-Length: 65

unknown image in /images/create?fromImage=postgres%3A11.2-alpine
```

### Information

After upgrading to Docker Desktop 4.13.0 (89412) with Docker Engine 20.10.20, downloading the `postgres:11.2` image using the [CreateImage](https://docs.docker.com/engine/api/v1.41/#tag/Image/operation/ImageCreate) operation in the Docker API and combining the image name and tag in the `fromImage` parameter is failing with the following error message:
```
unknown image in /images/create?fromImage=postgres%3A11.2
```

It's working when using the `fromImage` and `tag` parameters (which `docker pull` is doing):
```
# echo -e "POST /images/create?fromImage=postgres&tag=11.2-alpine HTTP/1.0\r\n" | nc -U /var/run/docker.sock
HTTP/1.0 200 OK
Api-Version: 1.41
Content-Type: application/json
Date: Thu, 27 Oct 2022 20:07:29 GMT
Docker-Experimental: false
Ostype: linux
Server: Docker/20.10.20 (linux)

{"status":"Pulling from library/postgres","id":"11.2-alpine"}
{"status":"Digest: sha256:ac1494b4cd5c0c6a85209e645fbb84ea186b55fa8c3678ef66dd9e4edf2b0195"}
{"status":"Status: Image is up to date for postgres:11.2-alpine"}
```

The issue cannot be reproduced on Linux (Debian Bullseye): https://github.com/moby/moby/issues/44366#issuecomment-1294000080

  - macOS Version: 13.0 (22A380) & 12.6 (21G115)
  - Intel chip or Apple chip: Apple M1
  - Docker Desktop Version: 4.13.0 (89412)

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`

```
# /Applications/Docker.app/Contents/MacOS/com.docker.diagnose check
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0014: are the backend processes running?
[PASS] DD0007: is the backend responding?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0012: is the VM networking working?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0019: is the com.docker.vmnetd process responding?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0032: do Docker networks overlap with host IPs?
No fatal errors detected.
```

### Steps to reproduce the behavior

```
# echo -e "POST /images/create?fromImage=postgres%3A11.2-alpine HTTP/1.0\r\n" | nc -U /var/run
# docker images postgres:11.2-alpine
```

**Comment by joschi** [comment]
Pulling via the image's digest works as expected:
```
# echo -e "POST /images/create?fromImage=postgres%40sha256%3Aac1494b4cd5c0c6a85209e645fbb84ea186b55fa8c3678ef66dd9e4edf2b0195 HTTP/1.0\r\n" | nc -U /v
ar/run/docker.sock
HTTP/1.0 200 OK
Api-Version: 1.41
Content-Type: application/json
Date: Wed, 02 Nov 2022 08:08:05 GMT
Docker-Experimental: false
Ostype: linux
Server: Docker/20.10.20 (linux)

{"status":"Pulling from library/postgres","id":"docker.io/library/postgres@sha256:ac1494b4cd5c0c6a85209e645fbb84ea186b55fa8c3678ef66dd9e4edf2b0195"}
{"status":"Digest: sha256:ac1494b4cd5c0c6a85209e645fbb84ea186b55fa8c3678ef66dd9e4edf2b0195"}
{"status":"Status: Image is up to date for postgres@sha256:ac1494b4cd5c0c6a85209e645fbb84ea186b55fa8c3678ef66dd9e4edf2b0195"}
```

**Comment by gesellix** [comment]
I ran into the same issue on Docker Desktop for Windows, see https://github.com/moby/moby/issues/44366#issuecomment-1304638977

**Comment by budarin** [comment]
the same on Windows and Mac

https://forums.docker.com/t/docker-registry-error-response-from-daemon-unknown-image-in-v1-41-images-create-fromimage/130944/6

**Comment by rimelek** [comment]
@budarin I don't think that is the same, since I could reproduce the issue with the registry but not with the postgres image. That's why I wrote there this was not the link I promised :) but it could be related. I could pull `postgres:11.2` on my Mac and on my Windows 11 without any problem.

**Comment by joschi** [comment]
> I could pull `postgres:11.2` on my Mac and on my Windows 11 without any problem.

Which version of Docker Desktop are you using? I could reproduce the issue on two different MacBooks with macOS 12.6 and 13.0.

**Comment by rimelek** [comment]
@joschi It was Docker Desktop 4.13.0 as it is in the title of this issue. Only after that I upgraded the Desktop to 4.13.1 and it worked with that too. On Windows 11 I tried only with 4.13.1 My Mac is macOS Ventura 13.0 with the M1 Apple chip.

**Comment by nicks** [comment]
Thanks for the report! We have some theories on what may cause this. Downgrading to DD 4.12 is probably the simplest fix right now.

**Comment by joschi** [comment]
Still failing on Docker Desktop 4.14.0.

```
# docker version
Client:
 Cloud integration: v1.0.29
 Version:           20.10.21
 API version:       1.41
 Go version:        go1.18.7
 Git commit:        baeda1f
 Built:             Tue Oct 25 18:01:18 2022
 OS/Arch:           darwin/arm64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.14.0 (91374)
 Engine:
  Version:          20.10.21
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.7
  Git commit:       3056208
  Built:            Tue Oct 25 17:59:41 2022
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.9
  GitCommit:        1c90a442489720eec95342e1789ee8a5e1b9536f
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

# echo -e "POST /images/create?fromImage=postgres%3A11.2 HTTP/1.0\r\n" | nc -U /var/run/docker.sock
HTTP/1.0 500 Internal Server Error
Content-Type: text/plain; charset=utf-8
X-Content-Type-Options: nosniff
Date: Fri, 11 Nov 2022 09:30:53 GMT
Content-Length: 58

unknown image in /images/create?fromImage=postgres%3A11.2
```

**Comment by maxcleme** [comment]
Closing this issue because a fix has been released in Docker Desktop 4.15.0. See [release notes](https://docs.docker.com/desktop/release-notes/) for more details.

**Comment by joschi** [comment]
@maxcleme Thanks! I can confirm everything is working in Docker Desktop for Mac 4.15.0 again. 👍
